psychopy/tests/test_misc/test_event.py

```diff
@@ -226,6 +227,20 @@ def setup_class(self):
         self.win = Window([128,128], monitor=mon, winType='pyglet', pos=[50,50], autoLog=False)
         assert pygame.display.get_init() == 0
 
+    def test_shutdown_key(self):
+        key = 'escape'
```

Comment:
> wouldn't this fail without a crash? I.e. setting it to a string instead of a list would cause if thisKey in prefs.general['shutdownKeys']: in _onPygletKey to loop through characters rather than list items. In other words: shouldn't the API here only allow to set prefs.general['shutdownKeys'] to be a list?

This is exactly the reason why I'm including this test :) This test will in fact only pass if `SystemExit` is raised, which will only happen if `core.quit()` is called, which in turn only happens (during this test) if our shutdown key was properly handled. See the next lines with the `pytest.raises()` context handler. I was also surprised it worked like this... So either the test is somehow broken or the behavior is actually what we really want! I can check more thoroughly tomorrow or on Tuesday.